packages/date-picker/src/DatePicker/DatePickerMenu/DatePickerMenu.tsx

```diff
@@ -92,8 +92,13 @@ export const DatePickerMenu = forwardRef<HTMLDivElement, DatePickerMenuProps>(
       if (isSameUTCMonth(newMonth, month)) {
         return;
       }
+
+      const isSameUTCMonthAsCurrentValue = isSameUTCMonth(newMonth, value);
+
       setDisplayMonth(newMonth);
-      const newHighlight = getNewHighlight(highlight, month, newMonth);
+      const newHighlight = isSameUTCMonthAsCurrentValue
+        ? (value as Date)
+        : getNewHighlight(highlight, month, newMonth);
```

Comment:
I'd say we should update `getNewHighlight` to take in `value` as an arg, and do this logic internally